libgraph/include/katana/GraphTopology.h

```diff
@@ -43,6 +45,59 @@ struct KATANA_EXPORT GraphTopologyTypes {
   using EdgeDestVec = NUMAArray<Node>;
   using PropIndexVec = NUMAArray<PropertyIndex>;
   using EntityTypeIDVec = NUMAArray<EntityTypeID>;
+  */
+
+  struct Node: public OpaqueIDLinear<Node, uint32_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct Edge: public OpaqueIDLinear<Edge, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct OutEdgeHandle: public OpaqueIDLinear<OutEdgeHandle, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
```

Comment:
What do you think about making the constructors of all these handle types private and `GraphTopology` to be their friend? No code outside of topology should be able to construct a handle.

Reply:
Good suggestion but let's leave it for near future work.